### PR TITLE
Replaced config.log and config.Rdata with config.yml

### PR DESCRIPTION
**Pull request by tscheypidi**
## :bird: Purpose of this PR :bird:

This PR improves the handling of settings in the model output folder. Instead of 2 files (a machine readable config.Rdata and a human readable config.log) we have now one unified file which serves both purposes. 

* replaced redundant files config.Rdata and config.log with config.yml
* removed confidential information from config before storing it in config.yml
 
## :wrench: Checklist for PR creator :wrench:

- [x] Labeling pull request correctly [from the label list](https://github.com/magpiemodel/magpie/labels).

* Low risk : Simple bugfixes (missing files, updated documentation, typos) or Start/output scripts
* Medium risk : New realization / Changes to existing realization / Other changes which don't modify default.cfg
* High risk : New input files (if cfg$input is changed in default.cfg) / Modification to core model (eg. changes in equations, calculations, introduction of new sets etc.) / Other changes in default.cfg

- [x] Providing additional information based on PR label

* Low risk : No new model run needed.
* Medium risk : Default run based on the current version of the fork from which PR is made
* High risk
  * Default run from the current develop branch
  * Default run based on the current version of the fork from which PR is made

- [x] :chart_with_downwards_trend: Performance loss/gain from current default behavior :chart_with_upwards_trend:
  * Current develop branch default : ** mins
  * This PR's default :  ** mins

- [x] Added changes to `CHANGELOG.md`
- [x] Compilation check (model starts without compilation errors - use `gams main.gms action=c` in model folder for testing).
- [x] No hard coded numbers and cluster/country/region names.
- [x] The new code doesn't contain declared but unused parameters or variables.
- [x] Where relevant, In-code comments added including documentation comments.
- [x] Made sure that documentation created with [`goxygen`](https://github.com/pik-piam/goxygen) is okay (use `goxygen::goxygen()` for testing).
- [x] Changes to [`magpie4`](https://github.com/pik-piam/magpie4) R library for post processing of model output (ideally backward compatible).
- [x] Self-review of my own code.
- [x] For high risk runs: validation of major model indicators - Land-use, emissions, food prices, Tau.  %Delete this line in case it is not a high risk run%

## :warning: Additional notes or warnings :warning:
NA

## :rotating_light: Checklist for RSE reviewer :rotating_light:

- [ ] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] No unnecessary increase in module interfaces
- [ ] All required updates in interfaces (if any) have been properly adressed in the module contracts
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

## :rotating_light: Checklist for MAgPIE reviewer :rotating_light:

- [ ] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] Changes to the model are scientifically sound
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

### CODE REVIEW

File: modules/42_water_demand/all_sectors_aug13/input.gms [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 scalars
 
-s42_watdem_nonagr_scenario         Scenario for non agricultural water demand from WATERGAP     (1)             / 1 /
+s42_watdem_nonagr_scenario         Scenario for non agricultural water demand from WATERGAP     (1)             / 2 /
```
Comment (dklein-pik) [inline modules/42_water_demand/all_sectors_aug13/input.gms:10]: Does it have any effect for users that this parameter changed from 1 to 2?

**Review by dklein-pik — APPROVED** [review]

File: modules/42_water_demand/all_sectors_aug13/input.gms (hunk 1 above)
Comment (tscheypidi) [reply modules/42_water_demand/all_sectors_aug13/input.gms:10]: good catch. No, it does not have an effect. It seems that there was just an inconsistency between the setting GAMS and in default.cfg. Now both show the same value (2).

**Review by abhimishr — APPROVED** [review]

File: scripts/output/extra/validation_exising_report.R [hunk 2]
```diff
@@ -12,6 +12,7 @@
```
Comment (dklein-pik) [inline scripts/output/extra/validation_exising_report.R:12]: Just caught my eye but not related to this PR: The name of this file is missing a "t" in "exising".

File: scripts/performance_test.R [hunk 3]
```diff
@@ -10,6 +10,7 @@ source("scripts/start_functions.R")
 
 performance_start <- function(cfg="default.cfg",modulepath="modules/",id="performance",sequential=NA) {
   require(lucode2)
+  require(yaml)
```
Comment (dklein-pik) [inline scripts/performance_test.R:13]: Why is this necessary?

File: scripts/performance_test.R [hunk 4]
```diff
@@ -116,7 +117,7 @@ performance_collect <- function(id="performance",results_folder="output/",plot=T
                        rows=ms["rows"],columns=ms["columns"],nonzeroes=ms["nonzeroes"],nlcode=ms["nlcode"],nlnonzeroes=ms["nlnonzeroes"])
     results <- rbind(results,tmp2)
   }
-  load(file.path(default,"config.Rdata"))
+  cfg <- gms::loadConfig(file.path(default, "config.yml"))
```
Comment (dklein-pik) [inline scripts/performance_test.R:120]: Why is no require(gms) necessary?

**Review by dklein-pik — APPROVED** [review]

File: scripts/performance_test.R (hunk 3 above)
Comment (tscheypidi) [reply scripts/performance_test.R]: because it is not ;) I removed it

File: scripts/performance_test.R (hunk 4 above)
Comment (tscheypidi) [reply scripts/performance_test.R:119]: when directly adressed via `gms::` the require statement is not necessary

File: scripts/output/extra/validation_exising_report.R (hunk 2 above)
Comment (tscheypidi) [reply scripts/output/extra/validation_exising_report.R:12]: fixed as well

File: scripts/performance_test.R (hunk 4 above)
Comment (dklein-pik) [reply scripts/performance_test.R:119]: Is it then necessary to add `library(gms)` as done in many other files, where also `gms::` is used?

Comment (tscheypidi) [reply scripts/performance_test.R:119]: not a `must` but a `can`. It can help to get an impression at the beginning of the file what dependencies are relevant
